### ISSUE TITLE
Synchronous network call when app enters foreground?

### ISSUE THREAD

**Issue by nolanw** [Awful/Awful.app#632]
### Steps to reproduce

1. Go to certain part of my office building where you appear to have a full-strength LTE connection but in practice all HTTP requests fail instantly. (OK, it's the washroom.)
2. Launch Awful.

### Expected result

Either the launch screen appears, or the snapshot made when entering the background appears, or the still-in-memory interface appears almost immediately.

### Observed result

The app icon darkens on Springboard and no input is accepted for a solid 5+ seconds.

### Notes

Awful 3.25, iPhone 7, iOS 10.3.3.

Issue title is my guess about what is happening: Awful does a synchronous network call on the main thread when entering the foreground, and the weird network state is tripping that up. I've only ever seen this behaviour when in the weird network state, but I guess it could be a coincidence.

After the absurdly long time entering the foreground, every network request fails instantly with "the internet connection is offline". If I tap airplane mode on then tap it back off, the data connection seems to fix itself. If I'm proactive enough to do this before attempting to launch Awful, there's no delay entering the foreground.

There was a recent request for a "don't check the pasteboard for a URL" setting because some people were observing massive delays as a blocking call to iCloud or whatever would accompany any inspection of the pasteboard. Maybe that's what I'm seeing here.

**Comment by realModusOperandi** [comment]
I don't think this is related to the pasteboard. With that one, we'd literally get a popup saying "Pasting from $MacNameHere." You could try toggling the setting, though...?

**Comment by nolanw** [comment]
I'm not seeing any kind of popup like that, good to know.

**Comment by nolanw** [comment]
Turned off the "check pasteboard for URL" setting and still had this issue, so that's ruled out.
